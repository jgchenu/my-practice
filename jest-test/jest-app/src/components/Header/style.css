.header-wrap {
  height: 100px;
  display: flex;
  align-items: center;
  box-sizing: border-box;
  padding-left: 100px;
  background-color: lightblue;
}

.header-wrap > input {
  outline: none;
  border: none;
  border-bottom: 1px solid #cccccc;
  height: 30px;
  font-size: 28px;
}
