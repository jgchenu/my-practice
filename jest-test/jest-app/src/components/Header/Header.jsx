import React, { useState } from "react";
import "./style.css";
function Header(props) {
  const [value, setValue] = useState("");
  return (
    <header className="header-wrap">
      <input
        type="text"
        data-test="input"
        value={value}
        placeholder="输入回车"
        onChange={(e) => setValue(e.target.value)}
        onKeyUp={(e) => {
          if (value && e.keyCode === 13) {
            props.addUndoItem(value);
          }
        }}
      />
    </header>
  );
}

export default Header;
