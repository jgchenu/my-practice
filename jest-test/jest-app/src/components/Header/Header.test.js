import React from "react";
import Header from "./Header";
import { shallow } from "enzyme";

test("Header 正常渲染", () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();
});

test("Header 组件应该渲染一个input", () => {
  const wrapper = shallow(<Header />);
  expect(wrapper.find('[data-test="input"]')).toExist();
});

test("Header 渲染的input初始值是空", () => {
  const wrapper = shallow(<Header />);
  const inputElem = wrapper.find('[data-test="input"]');
  expect(inputElem.prop("value")).toBe("");
});

test("Header 渲染的input change 事件会改变值", () => {
  const wrapper = shallow(<Header />);
  const userValue = "jgchen";
  const inputElem = wrapper.find('[data-test="input"]');
  inputElem.simulate("change", {
    target: { value: userValue },
  });
  expect(wrapper.find('[data-test="input"]').props().value).toBe(userValue);
});

test("Header 渲染的input 按下回车 没内容 回调事件不会被调用", () => {
  const wrapper = shallow(<Header />);
  const fn = jest.fn();
  const inputElem = wrapper.find('[data-test="input"]');
  inputElem.simulate("keyUp", {
    keyCode: 13,
  });
  expect(fn).not.toHaveBeenCalled();
});

test("Header 渲染的input 按下回车 有内容 回调事件会被调用", () => {
  const wrapper = shallow(<Header />);
  const fn = jest.fn();
  const inputElem = wrapper.find('[data-test="input"]');
  inputElem.simulate("change", {
    target: { value: "jgchen" },
  });
  inputElem.simulate("keyDown", {
    keyCode: 13,
  });
  expect(fn).not.toHaveBeenCalledWith("jgchen");
});
