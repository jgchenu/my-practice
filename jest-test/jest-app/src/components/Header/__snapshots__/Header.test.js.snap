// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 正常渲染 1`] = `
<header
  className="header-wrap"
>
  <input
    data-test="input"
    onChange={[Function]}
    onKeyUp={[Function]}
    placeholder="输入回车"
    type="text"
    value=""
  />
</header>
`;
